react/ProductName.tsx

```diff
@@ -138,6 +164,8 @@ function ProductNameWrapper(props: Props) {
   }
 
   const { product, selectedItem } = valuesFromContext
+  const displayMode =
+    props.displayMode !== undefined ? props.displayMode : 'plainText'
```

Comment:
You don't need this check here, you can just set a default value in the actual `ProductName` component.

I prefer having it there to do as little work as possible inside this `ProductNameWrapper` component. It makes this whole file and components easier to understand. Also, since you're just setting a default value by checking if something is `undefined`, you can just use JS in your favor and not have to do this check manually :)